README
NAME
    Net::Flickr::RDF - a.k.a RDF::Describes::Flickr

SYNOPSIS
     use Net::Flickr::RDF;
     use Config::Simple;
     use IO::AtomicFile;

     my $cfg = Config::Simple->new("/path/to/my.cfg");
     my $rdf = Net::Flickr::RDF->new($cfg);

     my $fh  = IO::AtomicFile->open("/foo/bar.rdf","w");

     $rdf->describe_photo({photo_id => 123,
                           secret   => 567,
                           fh       => \*$fh});

     $fh->close();

DESCRIPTION
    Describe Flickr photos as RDF.

    This package inherits from *Net::Flickr::API*.

OPTIONS
    Options are passed to Net::Flickr::Backup using a Config::Simple object
    or a valid Config::Simple config file. Options are grouped by "block".

  flickr
    * api_key
        String. *required*

        A valid Flickr API key.

    * api_secret
        String. *required*

        A valid Flickr Auth API secret key.

    * auth_token
        String. *required*

        A valid Flickr Auth API token.

  rdf
    * query_geonames
        Boolean.

        If true and a photo has geodata (latitude, longitude) associated
        with it, then the geonames.org database will be queried for a
        corresponding match. Data will be added as properties of the photo's
        geo:Point description. For example :

         <geo:Point rdf:about="http://www.flickr.com/photos/35034348999@N01/272880469#location">
            <geo:long>-122.025151</geo:long>
            <flickr:accuracy>16</ymaps:flickr>
            <acl:access>visbility</acl:access>
            <geo:lat>37.417839</geo:lat>
            <acl:accessor>public</acl:accessor>
            <geoname:Feature rdf:resource="http://ws.geonames.org/rdf?geonameId=5409655"/>
         </geo:Point>

         <geoname:Feature rdf:about="http://ws.geonames.org/rdf?geonameId=5409655">
            <geoname:featureCode>PPLX</geoname:featureCode>
            <geoname:countryCode>US</geoname:countryCode>
            <geoname:regionCode>CA</geoname:regionCode>
            <geoname:region>California</geoname:region>
            <geoname:city>Santa Clara</geoname:city>
            <geoname:gtopo30>2</geoname:gtopo30>
         </geoname:Feature>

        Default is false.

    * <query_trynt_color_api>
        Boolean.

        If true, the trynt colour extraction web service will be queried
        with the URL for the "medium" sized photo. Each colour will be added
        as it's own description, referenced from the photo's principal
        description. For example :

         <flickr:photo rdf:about="http://www.flickr.com/photos/35034348999@N01/299815039">
           <trynt:hasColor rdf:resource="http://www.flickr.com/photos/35034348999@N01/299815039#c0c0c0"/>
         </flickr:photo>

         <trynt:color rdf:about="http://www.flickr.com/photos/35034348999@N01/299815039#c0c0c0">
            <trynt:hexidecimal>c0c0c0</trynt:hexidecimal>
            <trynt:count>654</trynt:count>
         </trynt:color>

        Default is false.

PACKAGE METHODS
  __PACKAGE__->new($cfg)
    Where $cfg is either a valid *Config::Simple* object or the path to a
    file that can be parsed by *Config::Simple*.

    Returns a *Net::Flickr::RDF* object.

PACKAGE METHODS YOU MAY CARE ABOUT
  __PACKAGE__->build_photo_uri(\%data)
    Returns a URL as a string.

  __PACKAGE__->build_geo_uri(\%data)
  __PACKAGE__->build_user_tag_uri(\@data)
    Returns a URL as a string.

  __PACKAGE__->build_global_tag_uri(\@data)
    Returns a URL as a string.

  __PACKAGE__->build_machinetag_uri(\@data)
    Returns a URL as a string.

  __PACKAGE__->build_user_uri($user_id)
    Returns a URL as a string.

  __PACKAGE__->build_group_uri($group_id)
    Returns a URL as a string.

  __PACKAGE__->build_grouppool_uri($group_id)
    Returns a URL as a string.

  __PACKAGE__->build_photoset_uri(\%set_data)
    Returns a URL as a string.

  __PACKAGE__->prune_triples(\@triples)
    Removes duplicate triples from *@triples*. Returns an array reference.

OBJECT METHODS YOU SHOULD CARE ABOUT
  $obj->describe_photo(\%args)
    Valid arguments are :

    * photo_id
        Int. *required*

    * secret
        String.

    * fh
        File-handle.

        Default is STDOUT.

    Returns true or false.

OBJECT METHODS YOU MAY CARE ABOUT
  $obj->collect_photo_data($photo_id,$secret)
    Returns a hash ref of the meta data associated with a photo.

    If any errors are unencounter an error is recorded via the log method
    and the method returns undef.

  $obj->collect_group_data($group_id)
    Returns a hash ref of the meta data associated with a group.

    If any errors are unencounter an error is recorded via the log method
    and the method returns undef.

  $obj->collect_user_data($user_id)
    Returns a hash ref of the meta data associated with a user.

    If any errors are unencounter an error is recorded via the log method
    and the method returns undef.

  $obj->collect_photoset_data($photoset_id)
    Returns a hash ref of the meta data associated with a photoset.

    If any errors are unencounter an error is recorded via the log method
    and the method returns undef.

  $obj->collect_cc_data()
    Returns a hash ref of the Creative Commons licenses used by Flickr.

    If any errors are unencounter an error is recorded via the log method
    and the method returns undef.

  $obj->collect_comment_data()
    Returns a hash ref of comments made about a photo.

  $obj->make_photo_triples(\%data)
    Returns an array ref (or alist in a wantarray context) of array refs of
    the meta data associated with a photo (*%data*).

  $obj->make_user_triples(\%user_data)
    Returns an array ref (or list in a wantarray context) of array refs of
    the meta data associated with a user (*%user_data*).

  $obj->make_tag_triples(\@tag_data)
    Returns an array ref (or list in a wantarray context) of array refs of
    the meta data associated with a tag (*@tag_data*).

  $pkg->make_photoset_triples(\%set_data)
    Returns an array ref (or list in a wantarray context) of array refs of
    the meta data associated with a photoset (*%set_data*).

  $obj->make_geo_triples(\%geo_data)
  $obj->make_flickrplaces_triples(\%geo_data)
  $obj->make_geonames_triples(\%geo_data)
  $obj->make_group_triples(\%group_data)
    Returns an array ref (or list in a wantarray context) of array refs of
    the meta data associated with a group (*%group_data*).

  $obj->make_grouppool_triples(\%group_data)
    Returns an array ref (or list in a wantarray context) of array refs of
    the meta data associated with a group pool (*%group_data*).

  $obj->make_cc_triples($url)
    Returns an array ref (or list in a wantarray context) of array refs of
    the meta data associated with a Creative Commons license (*$url*).

  $obj->make_comment_triples(\%data)
    Returns an array ref (or alist in a wantarray context) of array refs of
    the meta data associated with a photo comment (*%data*).

  $obj->geodata_from_tags(\%data)
    Try to parse out geolocative data from a collection of tag data.

    Returns a hash ref (containing 'lat' and 'long' keys) on success or
    undef if there were no matches.

  $obj->namespaces()
    Returns a hash ref of the prefixes and namespaces used by
    *Net::Flickr::RDF*

    The default key/value pairs are :

    a   http://www.w3.org/2000/10/annotation-ns

    acl http://www.w3.org/2001/02/acls#

    atom
        http://www.w3.org/2005/Atom/

    cc  http://web.resource.org/cc/

    dc  http://purl.org/dc/elements/1.1/

    dcterms
        http://purl.org/dc/terms/

    exif
        http://nwalsh.com/rdf/exif#

    exifi
        http://nwalsh.com/rdf/exif-intrinsic#

    flickr
        x-urn:flickr:

    foaf
        http://xmlns.com/foaf/0.1/#

    geo http://www.w3.org/2003/01/geo/wgs84_pos#

    geoname
        http://www.geonames.org/onto#

    i   http://www.w3.org/2004/02/image-regions#

    rdf http://www.w3.org/1999/02/22-rdf-syntax-ns#

    rdfs
        http://www.w3.org/2000/01/rdf-schema#

    skos
        http://www.w3.org/2004/02/skos/core#

    trynt
        http://www.trynt.com#

    ymaps
        urn:yahoo:maps

  $obj->add_namespace($prefix, $namespace)
    Add a prefix and namespace URI to the list of known namespaces.

    This method returns a string containing the prefix you should use for
    the namespace URI passed. If the prefix passed to the method is already
    reserved for another namespace the method will return the following
    string:

     "nfr_" + I<the prefix passed to the method>

  $obj->namespace_prefix($uri)
    Return the namespace prefix for *$uri*

  $obj->uri_shortform($prefix,$name)
    Returns a string in the form of *prefix*:*property*. The property is the
    value of $name. The prefix passed may or may be the same as the prefix
    returned depending on whether or not the user has defined or redefined
    their own list of namespaces.

    Unless this package is subclassed the prefix passed to the method is
    assumed to be one of prefixes in the default list of namespaces.

  $obj->api_call(\%args)
    Valid args are :

    * method
        A string containing the name of the Flickr API method you are
        calling.

    * args
        A hash ref containing the key value pairs you are passing to
        *method*

    If the method encounters any errors calling the API, receives an API
    error or can not parse the response it will log an error event, via the
    log method, and return undef.

    Otherwise it will return a *XML::LibXML::Document* object (if
    XML::LibXML is installed) or a *XML::XPath* object.

  $obj->log()
    Returns a *Log::Dispatch* object.

  $obj->serialise_triples(\@triples,\*$fh)
    Print *@triples* as RDF/XML to a filehandle (*$fh*). If no filehandle is
    defined, prints to STDOUT.

  $obj->serialize_triples(\@triples,\*$fh)
    An alias for *serialise_triples*

VERSION
    2.01

DATE
    $Date: 2007/09/01 19:11:12 $

AUTHOR
    Aaron Straup Cope <ascope@cpan.org>

EXAMPLES
  CONFIG FILES
    This is an example of a Config::Simple file used to collect RDF data
    from Flickr

     [flickr] 
     api_key=asd6234kjhdmbzcxi6e323
     api_secret=s00p3rs3k3t
     auth_token=123-omgwtf4u

  RDF
    This is an example of an RDF dump for a photograph backed up from Flickr
    :

     <?xml version='1.0'?>    
     <rdf:RDF
      xmlns:geoname="http://www.geonames.org/onto#"
      xmlns:a="http://www.w3.org/2000/10/annotation-ns"
      xmlns:ph="http://www.machinetags.org/wiki/ph#camera"
      xmlns:filtr="http://www.machinetags.org/wiki/filtr#process"
      xmlns:nfr_geo="http://www.machinetags.org/wiki/geo#debug"
      xmlns:place="x-urn:flickr:place:"
      xmlns:exif="http://nwalsh.com/rdf/exif#"
      xmlns:trynt="http://www.trynt.com#"
      xmlns:mt="x-urn:flickr:machinetag:"
      xmlns:exifi="http://nwalsh.com/rdf/exif-intrinsic#"
      xmlns:geonames="http://www.machinetags.org/wiki/geonames#feature"
      xmlns:dcterms="http://purl.org/dc/terms/"
      xmlns:dc="http://purl.org/dc/elements/1.1/"
      xmlns:geo="http://www.w3.org/2003/01/geo/wgs84_pos#"
      xmlns:acl="http://www.w3.org/2001/02/acls#"
      xmlns:skos="http://www.w3.org/2004/02/skos/core#"
      xmlns:foaf="http://xmlns.com/foaf/0.1/"
      xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
      xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
      xmlns:flickr="x-urn:flickr:"
     >

      <flickr:user rdf:about="http://www.flickr.com/people/72238590@N00">
        <foaf:mbox_sha1sum>2fc2c76d7634d1a6446b1898bf5471205ed3d0cb</foaf:mbox_sha1sum>
        <foaf:name></foaf:name>
        <foaf:nick>thincvox</foaf:nick>
      </flickr:user>

      <flickr:tag rdf:about="http://www.flickr.com/photos/35034348999@N01/tags/filtr:process=filtr">
        <skos:altLabel>filtr</skos:altLabel>
        <dc:creator rdf:resource="http://www.flickr.com/people/35034348999@N01"/>
        <skos:broader rdf:resource="http://www.flickr.com/photos/tags/filtr:process=filtr"/>
        <skos:prefLabel rdf:resource="filtr:process=filtr"/>
      </flickr:tag>

      <geoname:Feature rdf:about="http://ws.geonames.org/rdf?geonameId=5400754">
        <geoname:featureCode>PPLX</geoname:featureCode>
        <geoname:countryCode>US</geoname:countryCode>
        <geoname:regionCode>CA</geoname:regionCode>
        <geoname:gtopo30>58</geoname:gtopo30>
        <geoname:region>State of California</geoname:region>
        <geoname:city>San Francisco County</geoname:city>
        <dc:isReferencedBy rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395"/>
      </geoname:Feature>

      <flickr:machinetag rdf:about="http://www.machinetags.org/wiki/geonames#feature">
        <mt:predicate>feature</mt:predicate>
        <mt:namespace>geonames</mt:namespace>
        <dc:isReferencedBy rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395"/>
      </flickr:machinetag>

      <dcterms:StillImage rdf:about="http://farm1.static.flickr.com/232/522214395_d2841bdbb0_o.jpg">
        <dcterms:relation>Original</dcterms:relation>
        <exifi:height>1944</exifi:height>
        <exifi:width>2592</exifi:width>
        <rdfs:seeAlso rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395"/>
        <rdfs:seeAlso rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#exif"/>
      </dcterms:StillImage>

      <flickr:tag rdf:about="http://www.flickr.com/photos/tags/cameraphone">
        <skos:prefLabel>cameraphone</skos:prefLabel>
      </flickr:tag>

      <flickr:tag rdf:about="http://www.flickr.com/photos/35034348999@N01/tags/filtr">
        <skos:prefLabel>filtr</skos:prefLabel>
        <dc:creator rdf:resource="http://www.flickr.com/people/35034348999@N01"/>
        <skos:broader rdf:resource="http://www.flickr.com/photos/tags/filtr"/>
      </flickr:tag>

      <rdf:Description rdf:about="http://www.flickr.com/photos/35034348999@N01/522214395#exif">
        <exif:flash>Flash did not fire, auto mode</exif:flash>
        <exif:digitalZoomRatio>100/100</exif:digitalZoomRatio>
        <exif:isoSpeedRatings>100</exif:isoSpeedRatings>
        <exif:pixelXDimension>2592</exif:pixelXDimension>
        <exif:apertureValue>297/100</exif:apertureValue>
        <exif:pixelYDimension>1944</exif:pixelYDimension>
        <exif:focalLength>5.6 mm</exif:focalLength>
        <exif:dateTimeDigitized>2007-05-30T15:10:01PDT</exif:dateTimeDigitized>
        <exif:colorSpace>sRGB</exif:colorSpace>
        <exif:fNumber>f/2.8</exif:fNumber>
        <exif:dateTimeOriginal>2007-05-30T15:10:01PDT</exif:dateTimeOriginal>
        <exif:shutterSpeedValue>4351/1000</exif:shutterSpeedValue>
        <exif:exposureTime>0.049 sec (49/1000)</exif:exposureTime>
      </rdf:Description>

      <trynt:color rdf:about="http://www.flickr.com/photos/35034348999@N01/522214395#604040">
        <trynt:hexidecimal>604040</trynt:hexidecimal>
        <trynt:count>472</trynt:count>
      </trynt:color>

      <rdf:Description rdf:about="x-urn:flickr:tag">
        <rdfs:subClassOf rdf:resource="http://www.w3.org/2004/02/skos/core#Concept"/>
      </rdf:Description>

      <trynt:color rdf:about="http://www.flickr.com/photos/35034348999@N01/522214395#000000">
        <trynt:hexidecimal>000000</trynt:hexidecimal>
        <trynt:count>4841</trynt:count>
      </trynt:color>

      <flickr:tag rdf:about="http://www.flickr.com/photos/35034348999@N01/tags/sanfrancisco">
        <skos:prefLabel>san francisco</skos:prefLabel>
        <skos:altLabel>sanfrancisco</skos:altLabel>
        <dc:creator rdf:resource="http://www.flickr.com/people/35034348999@N01"/>
        <skos:broader rdf:resource="http://www.flickr.com/photos/tags/sanfrancisco"/>
      </flickr:tag>

      <flickr:tag rdf:about="http://www.flickr.com/photos/tags/sanfrancisco">
        <skos:prefLabel>sanfrancisco</skos:prefLabel>
      </flickr:tag>

      <dcterms:StillImage rdf:about="http://farm1.static.flickr.com/232/522214395_ed16f959a2.jpg">
        <dcterms:relation>Medium</dcterms:relation>
        <exifi:height>375</exifi:height>
        <exifi:width>500</exifi:width>
        <dcterms:isVersionOf rdf:resource="http://farm1.static.flickr.com/232/522214395_d2841bdbb0_o.jpg"/>
        <rdfs:seeAlso rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395"/>
        <rdfs:seeAlso rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#exif"/>
      </dcterms:StillImage>

      <flickr:tag rdf:about="http://www.flickr.com/photos/tags/geonames:feature=5405296">
        <skos:altLabel>5405296</skos:altLabel>
        <skos:broader rdf:resource="http://www.machinetags.org/wiki/geonames#feature"/>
        <skos:prefLabel rdf:resource="geonames:feature=5405296"/>
      </flickr:tag>

      <flickr:tag rdf:about="http://www.flickr.com/photos/35034348999@N01/tags/cameraphone">
        <skos:prefLabel>cameraphone</skos:prefLabel>
        <dc:creator rdf:resource="http://www.flickr.com/people/35034348999@N01"/>
        <skos:broader rdf:resource="http://www.flickr.com/photos/tags/cameraphone"/>
      </flickr:tag>

      <flickr:photoset rdf:about="http://www.flickr.com/photos/35034348999@N01/sets/72157594459261101">
        <dc:description></dc:description>
        <dc:title>LOG (2007)</dc:title>
        <dc:creator rdf:resource="http://www.flickr.com/people/35034348999@N01"/>
      </flickr:photoset>

      <flickr:comment rdf:about="http://www.flickr.com/photos/straup/522214395/#comment72157600293655654">
        <dc:identifier>6065-522214395-72157600293655654</dc:identifier>
        <dc:created>2007-05-31T14:54:25</dc:created>
        <a:body>Kittens!</a:body>
        <dc:creator rdf:resource="http://www.flickr.com/people/35034348999@N01"/>
        <a:annotates rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395"/>
      </flickr:comment>

      <flickr:user rdf:about="http://www.flickr.com/people/35034348999@N01">
        <foaf:mbox_sha1sum>587a68f90c4030a9b0c7d8ca6ff8549a8b40e5cd</foaf:mbox_sha1sum>
        <foaf:name>Aaron Straup Cope</foaf:name>
        <foaf:nick>straup</foaf:nick>
      </flickr:user>

      <flickr:tag rdf:about="http://www.flickr.com/photos/35034348999@N01/tags/ph:camera=n95">
        <skos:altLabel>n95</skos:altLabel>
        <dc:creator rdf:resource="http://www.flickr.com/people/35034348999@N01"/>
        <skos:broader rdf:resource="http://www.flickr.com/photos/tags/ph:camera=n95"/>
        <skos:prefLabel rdf:resource="ph:camera=n95"/>
      </flickr:tag>

      <rdf:Description rdf:about="x-urn:flickr:comment">
        <rdfs:subClassOf rdf:resource="http://www.w3.org/2000/10/annotation-nsAnnotation"/>
      </rdf:Description>

      <flickr:comment rdf:about="http://www.flickr.com/photos/straup/522214395/#comment72157600295486776">
        <dc:identifier>6065-522214395-72157600295486776</dc:identifier>
        <dc:created>2007-06-01T00:19:05</dc:created>
        <a:body>here kitty, kitty, &lt;a href=&quot;http://thincvox.com/audio_recordings/meow.mp3&quot;&gt;meow&lt;/a&gt;</a:body>
        <dc:creator rdf:resource="http://www.flickr.com/people/72238590@N00"/>
        <a:annotates rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395"/>
      </flickr:comment>

      <flickr:tag rdf:about="http://www.flickr.com/photos/35034348999@N01/tags/geonames:feature=5405296">
        <skos:altLabel>5405296</skos:altLabel>
        <dc:creator rdf:resource="http://www.flickr.com/people/35034348999@N01"/>
        <skos:broader rdf:resource="http://www.flickr.com/photos/tags/geonames:feature=5405296"/>
        <skos:prefLabel rdf:resource="geonames:feature=5405296"/>
      </flickr:tag>

      <flickr:machinetag rdf:about="http://www.machinetags.org/wiki/filtr#process">
        <mt:predicate>process</mt:predicate>
        <mt:namespace>filtr</mt:namespace>
        <dc:isReferencedBy rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395"/>
      </flickr:machinetag>

      <flickr:machinetag rdf:about="http://www.machinetags.org/wiki/geo#debug">
        <mt:predicate>debug</mt:predicate>
        <mt:namespace>geo</mt:namespace>
        <dc:isReferencedBy rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395"/>
      </flickr:machinetag>

      <flickr:photo rdf:about="http://www.flickr.com/photos/35034348999@N01/522214395">
        <filtr:process>filtr</filtr:process>
        <nfr_geo:debug>namespace test</nfr_geo:debug>
        <acl:access>visbility</acl:access>
        <dc:title>Untitled #1180563722</dc:title>
        <ph:camera>n95</ph:camera>
        <dc:rights>All rights reserved.</dc:rights>
        <acl:accessor>public</acl:accessor>
        <dc:description></dc:description>
        <dc:created>2007-05-30T15:10:01-0700</dc:created>
        <dc:dateSubmitted>2007-05-30T15:18:39-0700</dc:dateSubmitted>
        <geonames:feature>5405296</geonames:feature>
        <dc:creator rdf:resource="http://www.flickr.com/people/35034348999@N01"/>
        <dc:subject rdf:resource="http://www.flickr.com/photos/35034348999@N01/tags/sanfrancisco"/>
        <dc:subject rdf:resource="http://www.flickr.com/photos/35034348999@N01/tags/ph:camera=n95"/>
        <dc:subject rdf:resource="http://www.flickr.com/photos/35034348999@N01/tags/geonames:feature=5405296"/>
        <dc:subject rdf:resource="http://www.flickr.com/photos/35034348999@N01/tags/geo:debug=namespacetest"/>
        <dc:subject rdf:resource="http://www.flickr.com/photos/35034348999@N01/tags/filtr:process=filtr"/>
        <dc:subject rdf:resource="http://www.flickr.com/photos/35034348999@N01/tags/filtr"/>
        <dc:subject rdf:resource="http://www.flickr.com/photos/35034348999@N01/tags/cameraphone"/>
        <dcterms:isPartOf rdf:resource="http://www.flickr.com/photos/35034348999@N01/sets/72157594459261101"/>
        <a:hasAnnotation rdf:resource="http://www.flickr.com/photos/straup/522214395/#comment72157600295486776"/>
        <a:hasAnnotation rdf:resource="http://www.flickr.com/photos/straup/522214395/#comment72157600293655654"/>
        <geo:Point rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#location"/>
        <trynt:hasColor rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#000000"/>
        <trynt:hasColor rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#c0c0c0"/>
        <trynt:hasColor rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#200000"/>
        <trynt:hasColor rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#402020"/>
        <trynt:hasColor rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#808080"/>
        <trynt:hasColor rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#a08080"/>
        <trynt:hasColor rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#000020"/>
        <trynt:hasColor rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#c0a0a0"/>
        <trynt:hasColor rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#604040"/>
        <trynt:hasColor rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#806060"/>
      </flickr:photo>

      <dcterms:StillImage rdf:about="http://farm1.static.flickr.com/232/522214395_ed16f959a2_t.jpg">
        <dcterms:relation>Thumbnail</dcterms:relation>
        <exifi:height>75</exifi:height>
        <exifi:width>100</exifi:width>
        <dcterms:isVersionOf rdf:resource="http://farm1.static.flickr.com/232/522214395_d2841bdbb0_o.jpg"/>
        <rdfs:seeAlso rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395"/>
        <rdfs:seeAlso rdf:resource="http://www.flickr.com/photos/35034348999@N01/522214395#exif"/>
      </dcterms:StillImage>

      <rdf:Description rdf:about="x-urn:flickr:machinetag">
        <rdfs:subClassOf rdf:resource="http://www.w3.org/2004/02/skos/core#Concept"/>
      </rdf:Description>

      <geo:Point rdf:about="http://www.flickr.com/photos/35034348999@N01/522214395#location">
        <geo:long>-122.401937</geo:long>
        <acl:access>visbility</acl:access>
        <geo:lat>37.794694</geo:lat>
        <flickr:accuracy>16</flickr:accuracy>
        <acl:accessor>public</acl:accessor>
        <skos:broader rdf:resource="http://ws.geonames.org/rdf?geonameId=5400754"/>
        <skos:broader rdf:resource="http://www.flickr.com/geo/United%20States/California/San%20Francisco/San%20Francisco"/>
      </geo:Point>

      <flickr:tag rdf:about="http://www.flickr.com/photos/tags/filtr:process=filtr">
        <skos:altLabel>filtr</skos:altLabel>
        <skos:broader rdf:resource="http://www.machinetags.org/wiki/filtr#process"/>
        <skos:prefLabel rdf:resource="filtr:process=filtr"/>
      </flickr:tag>

      <trynt:color rdf:about="http://www.flickr.com/photos/35034348999@N01/522214395#808080">
        <trynt:hexidecimal>808080</trynt:hexidecimal>
        <trynt:count>726</trynt:count>
      </trynt:color>

      <trynt:color rdf:about="http://www.flickr.com/photos/35034348999@N01/522214395#a08080">
        <trynt:hexidecimal>a08080</trynt:hexidecimal>
        <trynt:count>575</trynt:count>
      </trynt:color>

      <trynt:color rdf:about="http://www.flickr.com/photos/35034348999@N01/522214395#000020">
        <trynt:hexidecimal>000020</trynt:hexidecimal>
        <trynt:count>499</trynt:count>
      </trynt:color>

      <trynt:color rdf:about="http://www.flickr.com/photos/35034348999@N01/522214395#402020">
        <trynt:hexidecimal>402020</trynt:hexidecimal>
        <trynt:count>749</trynt:count>
      </trynt:color>

      <flickr:tag rdf:about="http://www.flickr.com/photos/tags/ph:camera=n95">
        <skos:altLabel>n95</skos:altLabel>

Changes
# -*-n3-*-

@prefix dc:      <http://purl.org/dc/elements/1.1/> .
@prefix dcterms: <http://purl.org/dc/terms/> .
@prefix doap:    <http://usefulinc.com/ns/doap#> .
@prefix asc:     <http://aaronland.info/ns/changefile/> .

<>
        dc:identifier           "$Id: Changes,v 1.43 2008/02/24 18:38:07 asc Exp $" ;
        dc:description          "Changes for Net::Flickr::RDF.pm" ;
        dcterms:references      <http://aaronland.info/perl/net/flickr/rdf/> .

<http://aaronland.info/perl/net/delicious/Net-Flickr-RDF-2.1.tar.gz>
        dcterms:isVersionOf <http://aaronland.info/perl/flickr/rdf/> ;
        dcterms:replaces    <http://aaronland.info/perl/net/flickr/rdf/Net-Flickr-RDF-2.01.tar.gz>; 

        doap:Version    [
                        doap:revision     "2.1" ;
                        doap:created      "2008-02-24" ;
                        ];
        asc:changes     [
                        asc:addition      "Capture Flickr Places information (old skool geo resources are linked via rdf:seeAlso)" ;
                        asc:update        "Require Net::Flickr::API 1.68" ;
                        ] . 

<http://aaronland.info/perl/net/delicious/Net-Flickr-RDF-2.01.tar.gz>
        dcterms:isVersionOf <http://aaronland.info/perl/flickr/rdf/> ;
        dcterms:replaces    <http://aaronland.info/perl/net/flickr/rdf/Net-Flickr-RDF-2.0.tar.gz>; 

        doap:Version    [
                        doap:revision     "2.01" ;
                        doap:created      "2007-09-03" ;
                        ];
        asc:changes     [
                        asc:update        "Require N:F:API 1.67" ;
                        ] . 

<http://aaronland.info/perl/net/delicious/Net-Flickr-RDF-2.0.tar.gz>
        dcterms:isVersionOf <http://aaronland.info/perl/flickr/rdf/> ;
        dcterms:replaces    <http://aaronland.info/perl/net/flickr/rdf/Net-Flickr-RDF-1.96.tar.gz>; 

        doap:Version    [
                        doap:revision     "2.0" ;
                        doap:created      "2007-06-02" ;
                        ];
        asc:changes     [
                        asc:addition      "Record machine tags as properties of the photo with pointers to machinetags.org" ;
                        asc:addition      "Record Flickr geo/place labels" ;
                        asc:addition      "Record versioning information about the RDF document itself" ;
                        asc:addition      "Added 'add_namespace' object method" ;
                        asc:update        "Changed the pointer for geonames:Feature (in geo:Point) to use skos:broader - THIS CHANGE IS NOT BACKWARDS COMPATIBLE" ;
                        asc:update        "Updated example RDF dump in POD" ;
                        asc:update        "Require Net::Flickr::API 1.65" ;
                        asc:bugfix        "Fix N3 in Changes file" ;
                        ] .
                                     
<http://aaronland.info/perl/net/delicious/Net-Flickr-RDF-1.96.tar.gz>
        dcterms:isVersionOf <http://aaronland.info/perl/flickr/rdf/> ;
        dcterms:replaces    <http://aaronland.info/perl/net/flickr/rdf/Net-Flickr-RDF-1.95.tar.gz>; 

        doap:Version    [
                        doap:revision     "1.96" ;
                        doap:created      "2006-11-19" ;
                        ];
        asc:changes     [
                        asc:update        "Replaced all calls to $xml->findvalue() with $xml->find()->string_value() to account for XML::XPath::Literal string overloading" ;
                        asc:update        "Updated to N:F:A 1.63" ;
                        ] .

<http://aaronland.info/perl/net/delicious/Net-Flickr-RDF-1.95.tar.gz>
        dcterms:isVersionOf <http://aaronland.info/perl/flickr/rdf/> ;
        dcterms:replaces    <http://aaronland.info/perl/net/flickr/rdf/Net-Flickr-RDF-1.94.tar.gz>; 

        doap:Version    [
                        doap:revision     "1.95" ;
                        doap:created      "2006-11-18" ;
                        ];
        asc:changes     [
                        asc:addition      "Added the option to query the Trynt web service to extract clour information" ;
                        asc:bugfix        "BLARGH! Fixed global tags URI (was /tags/TAG; is /tags/photos/TAG)" ;
                        ] .

<http://aaronland.info/perl/net/delicious/Net-Flickr-RDF-1.94.tar.gz>
        dcterms:isVersionOf <http://aaronland.info/perl/flickr/rdf/> ;
        dcterms:replaces    <http://aaronland.info/perl/net/flickr/rdf/Net-Flickr-RDF-1.93.tar.gz>; 

        doap:Version    [
                        doap:revision     "1.94" ;
                        doap:created      "2006-10-19" ;
                        ];
        asc:changes     [
                        asc:addition      "Query and store reverse geocoding and topo data for photos with lat/long information, using geonames.org" ;
                        asc:update        "Require Net::Flickr::API 1.62" ;
                        asc:update        "Use 'flickr' namespace for geo:Point/accuracy since Flickr measures accuracy differently than Y! Maps - this change is NOT BACKWARDS COMPATIBLE" ;
                        ] .

<http://aaronland.info/perl/net/delicious/Net-Flickr-RDF-1.93.tar.gz>
        dcterms:isVersionOf <http://aaronland.info/perl/flickr/rdf/> ;
        dcterms:replaces    <http://aaronland.info/perl/net/flickr/rdf/Net-Flickr-RDF-1.92.tar.gz>; 

        doap:Version    [
                        doap:revision     "1.93" ;
                        doap:created      "2006-09-07" ;
                        ];
        asc:changes     [
                        asc:update        "Better error handling in 'collect_user_data_by_screenname'" ;
                        asc:update        "Require Net::Flickr::API 1.61" ;
                        ] .

<http://aaronland.info/perl/net/delicious/Net-Flickr-RDF-1.92.tar.gz>
        dcterms:isVersionOf <http://aaronland.info/perl/flickr/rdf/> ;

Build.PL
# $Id: Build.PL,v 1.15 2007/09/03 05:32:48 asc Exp $

use strict;
use Module::Build;

my $build = Module::Build->new(module_name => 'Net::Flickr::RDF',
			       # dist_author
			       # dist_abstract
			       dist_version_from => 'lib/Net/Flickr/RDF.pm',
			       license           => 'perl',
			       requires => {
                                            'perl'               => '>= 5.8.0',
                                            'Net::Flickr::API'   => '>= 1.67',
                                            'RDF::Simple'        => '>= 0.22',
                                            'Date::Parse'        => '>= 2.27',
                                            'Date::Format'       => '>= 2.22',
                                            'Readonly'           => '0',
			       },
			       build_requires => {
				   'Test::Simple' => '>= 0.47',
			       },
			       create_readme      => 1,
			       create_makefile_pl => 'passthrough');

$build->create_build_script();
